watsononlinestore/watson_online_store.py

```diff
@@ -372,21 +373,115 @@ def get_watson_response(self, message):
         return response
 
     @staticmethod
-    def format_discovery_response(response):
-        """Takes specific ibm_store_html data and formats for Slack.
+    def format_discovery_response(response, data_source):
+        """Format data for Slack based on discovery data source.
 
-        This method is particular to the data in ibm_store_html and a new
-        method would be needed for other data. The formatted output is also
-        specific for Slack.
+        This method is handles the different data source data and formats
+        it specifically for Slack.
 
         :param dict response: input from Discovery
+        :param string data_source: The name of the discovery data source.
         :returns: cart_numer, name, url, image for each item returned
         :rtype: dict
         """
         output = []
         if not ('results' in response and response['results']):
             return output
 
+        def get_product_name(entry, data_source):
+            """ Pull product name from entry data for nice user display.
+
+            :param str data_source: The name of the discovery data source.
+            :returns: name of product
+            :rtype: str
+            """
+            product_name = ""
+
+            if data_source == "amazon":
+                # Pull out product name from enriched metadata.
+                if 'extracted_metadata' in entry:
+                    metadata = entry['extracted_metadata']
+                    if 'title' in metadata:
+                        product_name = metadata['title']
+            elif data_source == "ibm_store":
+                # Pull out product name from page text.
+                if 'text' in entry:
+                    product_tag = "Product:"
+                    category_tag = "Category:"
+                    text = entry['text']
+                    sidx = text.find(product_tag)
+                    if sidx > 0:
+                        sidx += len(product_tag)
+                        eidx = text.find(category_tag, sidx, len(text))
+                        if eidx > 0:
+                            product_name = text[sidx:eidx-1]
+
```

Comment:
If no data_source is set there's no results, so we probably should log an error.